third_party/move/move-model/src/builder/exp_builder.rs

```diff
@@ -4136,14 +4136,34 @@ impl<'env, 'translator, 'module_translator> ExpTranslator<'env, 'translator, 'mo
         context: &ErrorMessageContext,
     ) -> ExpData {
         // Translate arguments.
-        let (arg_types, translated_args) = self.translate_exp_list(args);
+        let (arg_types, mut translated_args) = self.translate_exp_list(args);
 
         // Special handling of receiver call functions
         if kind == CallKind::Receiver {
             debug_assert!(
                 module.is_none(),
                 "unexpected qualified name in receiver call"
             );
+            debug_assert!(
+                !args.is_empty(),
+                "receiver call needs to have at least one parameter"
+            );
+            let receiver_call_opt = self.get_receiver_function(&arg_types[0], name);
+            if let (Some(receiver_call), Exp_::ExpDotted(dotted)) =
```

Comment:
Can we add a test case where `receiver_call_opt` would be `None`? When I tried this out, we get a "unknown receiver function" error, but might be good to have a test to prevent any regressions.

Reply:
Added a test case to `calls_index_errors.move`.